src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1573,28 +1573,34 @@ void MergeTreeData::changeSettings(
                 StoragePolicyPtr new_storage_policy = global_context.getStoragePolicy(change.value.safeGet<String>());
                 StoragePolicyPtr old_storage_policy = getStoragePolicy();
 
-                checkStoragePolicy(new_storage_policy);
+                /// StoragePolicy is immutable!
```

Comment:
This comment seems a bit misleading, exactly why is StoragePolicy immutable?

Reply:
Hm, because it never changes but instead it is being constructed each time. I can remove that line if it's really misleading... :thinking: